sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -543,6 +709,65 @@ graph LR
+using a concept called graph _update_. This avoids a user having to rebuild and
+finalize a new executable graph when only the inputs & outputs to a graph
+change between submissions. Updates to a node take effect from the next
+submission of a graph and will not affect any previous submissions or in-flight
+executions of the same graph.
+
+The only type of node that is currently supported for updating in a graph is
+kernel execution nodes.
+
+The aspects of a kernel execution node that can be configured during update are:
+
+* Parameters to the kernel.
+* ND-Range of the kernel.
+
+Two methods are provided by the API to the user for performing this update.
+An API for updating the complete graph object, useful when the graph was
+created from a library, and an individual node update API.
+
+===== Whole Graph Update
+
+===== Individual Node Update
+
+Memory parameters to individual nodes in a graph in the `executable` state
+can be updated between graph executions using dynamic parameters. When a
+`dynamic_parameter` instance is created it is associated with a memory object.
+By registering the `dynamic_parameter` with a node using
+`node::register_dynamic_parameter()` the memory object underlying dynamic parameter
+is matched to a node argument, and the `dynamic_parameter` can then be
+used to update that node argument in future. This registration is necessary
+because SYCL kernel arguments don't have indices and ordering of lambda captures
+isn't defined, so to we need a way to tie the underlying node argument to
+a `dynamic_parameter` object.
+
+The update itself is then performed on the `dynamic_parameter` by calling
+`command_graph<graph_state::executable>::update_dynamic_parameter()`. If
+a user tries to register a dynamic parameter with a node, after that dynamic
+parameter has been updated with a new value, then it is the new value that
+must match one of the nodes arguments for the registration to be successful,
+rather than the value originally used to construct the dynamic parameter.
+
+The other node configuration that can be updated is the ND-Range, this can
+be set through `command_graph<graph_state::executable>::update_nd_range` but
+does not require any prior registration.
+
+Since the structure of the graph became fixed when finalizing, updating
+parameters on a node in a graph in the `executable` state will not change
+the already defined dependencies between nodes. This is important to note
+when updating buffer parameters to a node, since no edges will be automatically
+created or removed based on this change. Care should be taken that updates
+of buffer parameters do not change the behaviour of a graph when executed.
+
+For example, if there are two nodes (NodeA and NodeB) which are connected
+by an edge due to a dependency on the same buffer, both nodes must have
+this buffer parameter updated to the new value. This maintains the correct
+data dependency, and prevents unexpected behaviour.
```

Comment:
we may add one sentence if I understand correctly: This example also implies that one dynamic parameter can be registered to more than one nodes, and all the relative nodes to a given dynamic parameter should be registered to keep the correct data dependency.